server/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -599,7 +599,19 @@ public static Settings readSettingsFromStream(StreamInput in) throws IOException
 
     public static void writeSettingsToStream(Settings settings, StreamOutput out) throws IOException {
         // pull settings to exclude secure settings in size()
-        out.writeMap(settings.settings, StreamOutput::writeString, StreamOutput::writeGenericValue);
+        out.writeMap(settings.settings, StreamOutput::writeString, (streamOutput, value) -> {
+            if (value instanceof String) {
+                streamOutput.writeGenericString((String) value);
+            } else if (value instanceof List<?>) {
+                @SuppressWarnings("unchecked")
+                // exploit the fact that we know all lists to be string lists
+                final List<String> stringList = (List<String>) value;
+                streamOutput.writeGenericList(stringList, StreamOutput::writeGenericString);
+            } else {
+                assert value == null : "unexpected value [" + value + "]";
+                streamOutput.writeGenericNull();
+            }
```

Comment:
Looks like `Settings::fromXContent` also handles numbers and booleans. Is that effectively dead code that could be removed?